src/video.cpp

```diff
@@ -454,9 +447,8 @@ void VideoSystem::MainLoop()
 {
 	static const uint32 FRAME_DELAY = 30; // Number of milliseconds between two frames.
 	bool missing_sprites_check = false;
-	_finish = false;
 
-	while (!_finish) {
+	while (_game_control.running) {
```

Comment:
I wonder if this would be better written as for (;;) {
and exiting only at line 464.